pkg/transport/generic_transport_controller.go

```diff
@@ -438,6 +435,31 @@ func (c *genericTransportController) initializeWrappedObject(ctx context.Context
 	return wrappedObject, nil
 }
 
+func (c *genericTransportController) propagateWrappedObjectToClusters(ctx context.Context, desiredWrappedObject *unstructured.Unstructured,
+	currentWrappedObjectList *unstructured.UnstructuredList, destinations []v1alpha1.Destination) error {
+	// if the desired wrapped object is nil, that means we should not propagate this object.
+	// this may happen when the workload section is empty.
+	// this is not an error state but a valid scenario.
+	// return without propagating, the delete section will remove existing instances of the wrapped object from all current destinations.
+	if desiredWrappedObject == nil {
```

Comment:
Wouldn't ::initializeWrappedObject return an empty wrapped object, and not a nil pointer? That is according to transport's WrapObjects documentation:
> WrapObjects gets slice of objects and wraps them into a single wrapped object. In case slice is empty, the function should return an empty wrapped object.